src/main/java/com/synopsys/integration/jenkins/scan/bridge/BridgeDownloadManager.java

```diff
@@ -1,100 +1,163 @@
-                String version = content.matches("^[0-9]+\\.[0-9]+\\.[0-9]+") ? content : null;
-                if (version != null) {
-                    availableVersions.add(version);
-                }
+                Utility.removeFile(versionFilePath, workspace, listener);
+
+                return latestVersion;
             }
-        } catch (Exception e) {
-            e.printStackTrace();
+            else {
+                return ApplicationConstants.NOT_AVAILABLE;
+            }
+        }
+        else {
+            return extractedVersionNumber;
         }
+    }
 
-        return availableVersions;
+    public String downloadVersionFile(String directoryUrl) {
+        String versionFileUrl = String.join("/", directoryUrl, ApplicationConstants.VERSION_FILE);
+        String tempVersionFilePath = null;
+
+        try {
+            FilePath tempFilePath = workspace.createTempFile("versions", ".txt");
+            URL url = new URL(versionFileUrl);
 
+            tempFilePath.copyFrom(url);
+            tempVersionFilePath = tempFilePath.getRemote();
+        } catch (IOException | InterruptedException e) {
+            listener.getLogger().printf(LogMessages.EXCEPTION_OCCURRED_WHILE_DOWNLOADING_VERSION_FILE, e.getMessage());
+        }
+        return tempVersionFilePath;
     }
 
-    public String getLatestVersion(List<String> versions) {
-        String latestVersion = "0.0.0";
+    public boolean versionFileAvailable(String directoryUrl) {
+        try {
+            URL url = new URL(String.join("/",directoryUrl,ApplicationConstants.VERSION_FILE));
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+            connection.setRequestMethod("HEAD");
+            return (connection.getResponseCode() >= 200 && connection.getResponseCode() < 300);
+        } catch (IOException e) {
+            listener.getLogger().printf(LogMessages.EXCEPTION_OCCURRED_WHILE_CHECKING_VERSION_FILE_AVAILABILITY, e.getMessage());
+            return false;
+        }
+    }
 
-        for (String version : versions) {
-            if (version.compareTo(latestVersion) > 0) {
-                latestVersion = version;
+    public String getDirectoryUrl(String downloadUrl) {
+        String directoryUrl = null;
+        try {
+            URI uri = new URI(downloadUrl);
+            String path = uri.getPath();
+
+            if (path.endsWith("/")) {
+                path = path.substring(0, path.length() - 1);
             }
+
+            String directoryPath = path.substring(0, path.lastIndexOf('/'));
+            directoryUrl = uri.getScheme().concat("://").concat(uri.getHost()).concat(directoryPath);
+        } catch (URISyntaxException e) {
+            listener.getLogger().printf(LogMessages.EXCEPTION_OCCURRED_WHILE_GETTING_DIRECTORY_URL_FROM_DOWNLOAD_URL, e.getMessage());
```

Comment:
Why are we logging message of exception as well? I believe, we need to continue even if the directory url is not found.
So can we think of logging only a standard message from our side.

Reply:
1) We are logging exception message to figure out in case something goes wrong.
2) We are proceeding with the execution flow if exception happens here. This segment is used to get the `versions.txt` url so that we can compare and decide whether we should download the new bridge binary or not.
3) Can you explain what you meant by **'standard message from our side'**?

Reply:
We will refactor the logging similar to Detect Plugin